src/Symfony/Component/Form/Tests/Extension/Validator/Constraints/FormValidatorTest.php

```diff
@@ -349,13 +348,27 @@ function () { throw new TransformationFailedException(); }
     public function testHandleCallbackValidationGroups()
     {
         $object = $this->getMock('\stdClass');
+        $options = array('validation_groups' => new GroupSequence(array('group1', 'group2')));
+        $form = $this->getBuilder('name', '\stdClass', $options)
+            ->setData($object)
+            ->getForm();
+
+        $this->expectValidateAt(0, 'data', $object, new GroupSequence(array('group1', 'group2')));
+
+        $this->validator->validate($form, new Form());
+
+        $this->assertNoViolation();
+    }
+
+    public function testHandleGroupSequenceValidationGroups()
+    {
+        $object = $this->getMock('\stdClass');
         $options = array('validation_groups' => array($this, 'getValidationGroups'));
         $form = $this->getBuilder('name', '\stdClass', $options)
             ->setData($object)
             ->getForm();
 
-        $this->expectValidateAt(0, 'data', $object, 'group1');
-        $this->expectValidateAt(1, 'data', $object, 'group2');
+        $this->expectValidateAt(0, 'data', $object, array('group1', 'group2'));
```

Comment:
Looks like this bug fix is breaking BC with this behavioral change. But it seems the code was broken so I'm still 👍 for this one.